backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/nodes/ReferencingPerformActionUsageNodeDescriptionProvider.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.diagram.general.view.nodes;
+
+import java.util.List;
+
+import org.eclipse.sirius.components.view.builder.IViewDiagramElementFinder;
+import org.eclipse.sirius.components.view.builder.providers.IColorProvider;
+import org.eclipse.sirius.components.view.diagram.NodeDescription;
+import org.eclipse.sirius.components.view.diagram.NodeToolSection;
+import org.eclipse.syson.diagram.common.view.services.description.ReferencingPerformActionUsageNodeDescriptionService;
+import org.eclipse.syson.sysml.PerformActionUsage;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.eclipse.syson.util.AQLUtils;
+
+/**
+ * Used to create the "referencing" PerformActionUsage node description in the General View diagram.<br>
+ * There are two separate node descriptions mapped to {@link PerformActionUsage}:<br>
+ * - {@link ReferencingPerformActionUsageNodeDescriptionProvider}: when the performed action is a separated action
+ * usage<br>
+ * - {@link PerformActionUsageNodeDescriptionProvider}: when the performed action is the perform action itself.
+ *
+ * @author Jerome Gout
+ */
+public class ReferencingPerformActionUsageNodeDescriptionProvider extends UsageNodeDescriptionProvider {
```

Comment:
This class looks like to the new one declared in action-flow-view.
What don't you just have declared only one in common-view instead?

Reply:
I cannot share this class among all diagrams because each diagram has a specific `UsageNodeDescriptionProvider` which is its base class.